### PR TITLE
feat: Update keycloak  from 6.0.1 to 15

### PR DESCRIPTION
**Pull request by AndrienkoAleksandr**
### What does this PR do?
Update keycloak  from 6.0.1 to 15

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/19585

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [Custom resource definition file is up to date](https://github.com/eclipse-che/che-operator/blob/main/README.md#updating-custom-resource-definition-file)
- [x] [OLM bundles are up to date](https://github.com/eclipse-che/che-operator/blob/main/README.md#update-olm-bundle)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Review by mmorhun — APPROVED** [review]

**Comment by AndrienkoAleksandr** [comment]
ci/prow/v7-devworkspace-happy-path failed, but it doesn't relate to keycloak. I see in the log workspace created by devworkspace operator failed to start:

```
"DevWorkspace failed to start: Container maven has state ImagePullBackOff","Request.Namespace":"user-che","Request.Name":"spring-petclinic-qe2w","devworkspace_id":"workspacee6271cdf4b3d4116"}
```

**Review by tolusha — APPROVED** [review]
Pls. sounds good. Pls remove custom images before mergin.

File: .github/bin/common.sh [hunk 1]
```diff
@@ -449,23 +449,23 @@ login() {
...
+#   # Get the ocp domain for che custom resources
+#   export DOMAIN=$(oc get dns cluster -o json | jq .spec.baseDomain | sed -e 's/^"//' -e 's/"$//')
 
-  # Related issue:https://github.com/eclipse/che/issues/17681
-    cat >/tmp/che-cr-patch.yaml <<EOL
-spec:
-  server:
-    nonProxyHosts: oauth-openshift.apps.$DOMAIN
-EOL
+#   # Related issue:https://github.com/eclipse/che/issues/17681
```
Comment (flacatus) [inline .github/bin/common.sh:455]: Can we just remove the code if is not anymore need it?

**Review by tolusha — APPROVED** [review]

**Pull request merged**